docs/tutorials/talk-to-kafka.rst

```diff
@@ -0,0 +1,715 @@
+    schema_registry_conf = {
+        "url": f"{SCHEMA_REGISTRY_URI}"
+    }
+    schema_registry_client = SchemaRegistryClient(schema_registry_conf)
+
+    # Define the Avro schema for the message
+    value_str = """
+    {
+    "namespace": "example.avro",
+    "type": "record",
+    "name": "User",
+    "fields": [
+        {"name": "name", "type": "string"},
+        {"name": "age",  "type": ["int", "null"]}
+    ]
+    }
+    """
+
+    key_str = """
+    {
+    "namespace": "example.avro",
+    "type": "record",
+    "name": "Id",
+    "fields": [
+        {"name": "id", "type": "int"}
+    ]
+    }
+    """
+
+    # Define a class for the User object
+    class User(object):
+        def __init__(self, name, age):
+            self.name = name
+            self.age = age
+
+    # Define a class for the Key object
+    class Key(object):
+        def __init__(self, id):
+            self.id = id
+
+    # Converting dict to User object for AvroDeserializer function
+    def dict_to_user(obj, ctx):
+        return User(name=obj['name'],
+                    age=obj['age']
+        )
+
+    # Converting dict to Key object for AvroDeserializer function
+    def dict_to_key(obj, ctx):
+        return Key(id=obj["id"])
+
+    # Create the Avro deserializer
+    avro_deserializer_value = AvroDeserializer(
+        schema_registry_client, value_str, dict_to_user
+    )
+
+    avro_deserializer_key = AvroDeserializer(
+        schema_registry_client, key_str, dict_to_key
+    )
+
+    # Create the Kafka consumer
+    consumer = DeserializingConsumer(
+        conf
+    )
+
+    # Subscribe to the Kafka topic
+    consumer.subscribe(['demo-topic'])
+
+    # Consume messages from the Kafka topic
+    while True:
+        try:
+            msg = consumer.poll(1.0)
+
+            if msg is None:
+                continue
+
+            if msg.error():
+                print("Consumer error: {}".format(msg.error()))
+                continue
+            
+            user = avro_deserializer_value(
+                msg.value(), SerializationContext(msg.topic(), MessageField.VALUE)
+            )
+            key = avro_deserializer_key(
+                msg.key(), SerializationContext(msg.topic(), MessageField.KEY)
+            )
+
+            if user is not None and key is not None:
+                print(
+                    "Key --> {}\n"
+                    "User record --> name: {},"
+                    " age: {}\n".format(key.id, user.name, user.age)
+                )
+
+        except KeyboardInterrupt:
+                break
+
+        # Commit the offset for the message
+        consumer.commit()
+
+Observation
+~~~~~~~~~~~
+
+Once you execute the consumer program, you should see the following output on the second terminal while the consumer program stays running until you hit ``ctrl+C``:
+
+.. code:: shell
+
+    Key --> 1
+    User record --> name: John, age: 30
+
+Here's an overview of what each program does:
+
+Producer program
```

Comment:
Put the content in this section (as in: copy and paste it) in L#380 where you actually define the producer

Reply:
With all the commits, the line# got all mixed up. This explanation is for the Avro part of the consumer/producer and is under the **Observation** section. The format is _I'm writing some code_ and then _here's what the code does_. 

Do you suggest I duplicate these texts to some places within the file? if yes, could you please comment the new line numbers? Thanks.